zcash_client_backend/src/keys.rs

```diff
@@ -536,19 +537,46 @@ impl UnifiedFullViewingKey {
         self.orchard.as_ref()
     }
 
-    /// Attempts to derive the Unified Address for the given diversifier index.
+    /// Attempts to derive the Unified Address for the given diversifier index and
+    /// receiver types.
     ///
     /// Returns `None` if the specified index does not produce a valid diversifier.
-    // TODO: Allow filtering down by receiver types?
-    pub fn address(&self, j: DiversifierIndex) -> Option<UnifiedAddress> {
-        let sapling = if let Some(extfvk) = self.sapling.as_ref() {
+    pub fn address(
+        &self,
+        j: DiversifierIndex,
+        receiver_types: &[Typecode],
+    ) -> Option<UnifiedAddress> {
+        #[cfg(feature = "orchard")]
+        let orchard = {
+            let orchard_j = orchard::keys::DiversifierIndex::from(*j.as_bytes());
```

Comment:
This reminded me that we still need to migrate `orchard` to use `zip32` (which would remove this line). Opened https://github.com/zcash/orchard/issues/410.